src/System.Private.CoreLib/src/System/Delegate.cs

```diff
@@ -579,48 +573,13 @@ internal static Delegate CreateDelegateNoSecurityCheck(Type type, object target,
             return d;
         }
 
-        // Caution: this method is intended for deserialization only, no security checks are performed.
-        internal static Delegate CreateDelegateNoSecurityCheck(RuntimeType type, object firstArgument, MethodInfo method)
-        {
-            // Validate the parameters.
-            if (type == null)
-                throw new ArgumentNullException(nameof(type));
-            if (method == null)
-                throw new ArgumentNullException(nameof(method));
-
-
-            RuntimeMethodInfo rtMethod = method as RuntimeMethodInfo;
-            if (rtMethod == null)
-                throw new ArgumentException(SR.Argument_MustBeRuntimeMethodInfo, nameof(method));
-
-            if (!type.IsDelegate())
-                throw new ArgumentException(SR.Arg_MustBeDelegate, nameof(type));
-
-            // This API is used by the formatters when deserializing a delegate.
-            // They pass us the specific target method (that was already the
-            // target in a valid delegate) so we should bind with the most
-            // relaxed rules available (the result will never be ambiguous, it
-            // just increases the chance of success with minor (compatible)
-            // signature changes). We explicitly skip security checks here --
-            // we're not really constructing a delegate, we're cloning an
-            // existing instance which already passed its checks.
-            Delegate d = CreateDelegateInternal(type, rtMethod, firstArgument,
-                                              DelegateBindingFlags.SkipSecurityChecks |
-                                              DelegateBindingFlags.RelaxedSignature);
-
-            if (d == null)
-                throw new ArgumentException(SR.Arg_DlgtTargMeth);
-
-            return d;
-        }
-
         // V1 API.
-        public static Delegate CreateDelegate(Type type, MethodInfo method)
+        public static Delegate? CreateDelegate(Type type, MethodInfo method)
```

Comment:
@danmosemsft, why is this one still returning `Delegate?`?

Reply:
Assuming you agree it should be Delegate not Delegate?, I'll fix it in a PR I'm about to put up that relies on it.

Reply:
Yes, thank you.